profiler/src/ProfilerEngine/Datadog.Profiler.Native/AllocationsProvider.cpp

```diff
@@ -71,7 +64,13 @@ void AllocationsProvider::OnAllocation(uint32_t allocationKind,
     rawSample.ThreadInfo = threadInfo;
     threadInfo->AddRef();
     rawSample.AllocationSize = objectSize;
-    rawSample.AllocationClass = std::move(shared::ToString(shared::WSTRING(typeName)));
+
+    // The provided type name contains the metadata-based `xx syntax for generics instead of <>
+    // So rely on the frame store to get a C#-like representation like what is done for frames
+    if (!_pFrameStore->GetTypeName(classId, rawSample.AllocationClass))
+    {
+        rawSample.AllocationClass = std::move(shared::ToString(shared::WSTRING(typeName)));
```

Comment:
Why calling move here?
From what I see, the return value of `shared::ToString` is a rvalue-ref  so no need to call move on it

Reply:
✅ done